### PR TITLE
Search with Facet Parameters

### PR DESCRIPTION
**Pull request by ankitgoyalgloballogic**

### CODE REVIEW

File: AlfrescoContent/AlfrescoContent/Classes/Swaggers/Models/GenericBucketBucketInfo.swift [hunk 1]
```diff
@@ -15,13 +15,13 @@ public struct GenericBucketBucketInfo: Codable {
     /** The start of range */
     public var start: String?
     /** Includes values greater or equal to \&quot;start\&quot; */
-    public var startInclusive: Bool?
+    public var startInclusive: String?
```
Comment (aman-alfresco) [inline AlfrescoContent/AlfrescoContent/Classes/Swaggers/Models/GenericBucketBucketInfo.swift:18]: Have you pick this variable from Adf json?

File: AlfrescoContent/AlfrescoContent/Classes/Swaggers/Models/SearchRequest.swift [hunk 2]
```diff
@@ -22,18 +22,18 @@ public struct SearchRequest: Codable {
...
     public var pivots: [RequestPivot]?
     public var stats: [RequestStats]?
     public var spellcheck: RequestSpellcheck?
     public var scope: RequestScope?
     public var limits: RequestLimits?
     public var highlight: RequestHighlight?
     public var ranges: [RequestRange]?
+    public var facetQueries: [Queries]?
+    public var facetFields: Dictionary<String, [Fields]?>?
+    public var facetIntervals: Dictionary<String, [Intervals]?>?
```
Comment (aman-alfresco) [inline AlfrescoContent/AlfrescoContent/Classes/Swaggers/Models/SearchRequest.swift:34]: facet intervals have sets array, where are you passing it?

File: AlfrescoContent/AlfrescoContent/Classes/Swaggers/Models/GenericBucketBucketInfo.swift (hunk 1 above)
Comment (ankitgoyalgloballogic) [reply AlfrescoContent/AlfrescoContent/Classes/Swaggers/Models/GenericBucketBucketInfo.swift:18]: These are available in the response of search API.

File: AlfrescoContent/AlfrescoContent/Classes/Swaggers/Models/SearchRequest.swift (hunk 2 above)
Comment (ankitgoyalgloballogic) [reply AlfrescoContent/AlfrescoContent/Classes/Swaggers/Models/SearchRequest.swift:34]: These are in the Intervals array.

File: AlfrescoContent/AlfrescoContent/Classes/Swaggers/Models/SearchRequest.swift [hunk 3]
```diff
@@ -44,18 +44,25 @@ public struct SearchRequest: Codable {
...
         self.pivots = pivots
         self.stats = stats
         self.spellcheck = spellcheck
         self.scope = scope
         self.limits = limits
         self.highlight = highlight
         self.ranges = ranges
+        
+        self.facetQueries = facetQueries?.queries
+        if let facetFields = facetFields {
```
Comment (gl-lovekesh) [inline AlfrescoContent/AlfrescoContent/Classes/Swaggers/Models/SearchRequest.swift:56]: reduce the line of code as below
self.facetIntervals = nil
if let facetIntervals = facetIntervals {
            self.facetIntervals = ["intervals": facetIntervals.intervals]
  }

**Review by gl-lovekesh — COMMENTED** [review]
Fix the changes as per comment.

**Review by gl-lovekesh — APPROVED** [review]

**Pull request merged**